packages/alfa-network/test/network.spec.ts

```diff
@@ -0,0 +1,241 @@
+import { test } from "@siteimprove/alfa-test";
+
+import { Network } from "../src/network";
+
+// foo
+// |- 1 => bar
+// |- 2 => baz
+//    |- 3 => foo
+//       |- ...
+const network = Network.from<string, number>([
+  [
+    "foo",
+    [
+      ["bar", [1]],
+      ["baz", [2]],
+    ],
+  ],
+  ["bar", []],
+  ["baz", [["foo", [3]]]],
+]);
+
+test(".from() excludes connections with no edges", (t) => {
+  t.deepEqual(
+    Network.from<string, void>([["foo", [["bar", []]]]]).toArray(),
+    [["foo", []]]
+  );
+});
+
+test("#connect() connects two nodes in a network", (t) => {
+  // foo
+  // |- 1 => bar
+  //    |- 4 => baz
+  //       |- ...
+  // |- 2 => baz
+  //    |- 3 => foo
+  //       |- ...
+  t.deepEqual(network.connect("bar", "baz", 4).toArray(), [
+    ["baz", [["foo", [3]]]],
+    [
+      "foo",
+      [
+        ["baz", [2]],
+        ["bar", [1]],
+      ],
+    ],
+    ["bar", [["baz", [4]]]],
+  ]);
+
+  // foo
+  // |- 1 => bar
+  // |- 2 => baz
+  //    |- 3 => foo
+  //       |- ...
+  // |- 4 => baz
+  //    |- ...
+  t.deepEqual(network.connect("foo", "baz", 4).toArray(), [
+    ["baz", [["foo", [3]]]],
+    [
+      "foo",
+      [
+        ["baz", [4, 2]],
+        ["bar", [1]],
+      ],
+    ],
+    ["bar", []],
+  ]);
+});
+
+test("#disconnect() disconnects two nodes in a network", (t) => {
+  // foo
+  // |- 2 => baz
+  //    |- 3 => foo
+  //       |- ...
+  // bar
+  t.deepEqual(network.disconnect("foo", "bar").toArray(), [
+    ["baz", [["foo", [3]]]],
+    ["foo", [["baz", [2]]]],
+    ["bar", []],
```

Comment:
It's a bit weird that `.from` does not allow to create disjoint graphs, but `.disconnect` is. But I guess it wouldn't be clear which bit to drop.

Reply:
`Network.from()` can do that as well:

```ts
Network.from([
  ["foo", []],
  ["bar", []],
]);
```